plugin/src/main/java/com/epam/reportportal/extension/slack/SlackPluginExtension.java

```diff
@@ -40,110 +53,160 @@
+    initListeners();
+    initScripts();
+  }
+
+  private void initListeners() {
+    ApplicationEventMulticaster applicationEventMulticaster = applicationContext.getBean(
+        AbstractApplicationContext.APPLICATION_EVENT_MULTICASTER_BEAN_NAME,
+        ApplicationEventMulticaster.class
+    );
+    applicationEventMulticaster.addApplicationListener(pluginLoadedListener.get());
+    applicationEventMulticaster.addApplicationListener(launchFinishEventListenerSupplier.get());
+  }
 
-    private Map<String, CommonPluginCommand<?>> getCommonCommands() {
-        HashMap<String, CommonPluginCommand<?>> pluginCommands = new HashMap<>();
-        return pluginCommands;
+  private void initScripts() throws IOException {
+    try (Stream<Path> paths = Files.list(Paths.get(resourcesDir, SCRIPTS_DIR))) {
+      FileSystemResource[] scriptResources =
+          paths.sorted().map(FileSystemResource::new).toArray(FileSystemResource[]::new);
+      ResourceDatabasePopulator resourceDatabasePopulator =
+          new ResourceDatabasePopulator(scriptResources);
+      resourceDatabasePopulator.execute(dataSource);
     }
+  }
+
+  @Override
+  public void destroy() {
+    removeListeners();
+  }
+
+  private void removeListeners() {
+    ApplicationEventMulticaster applicationEventMulticaster = applicationContext.getBean(
+        AbstractApplicationContext.APPLICATION_EVENT_MULTICASTER_BEAN_NAME,
+        ApplicationEventMulticaster.class
+    );
+    applicationEventMulticaster.removeApplicationListener(pluginLoadedListener.get());
+    applicationEventMulticaster.removeApplicationListener(launchFinishEventListenerSupplier.get());
+  }
+
+  @Override
+  public Map<String, ?> getPluginParams() {
+    Map<String, Object> params = new HashMap<>();
+    params.put(ALLOWED_COMMANDS, new ArrayList<>(pluginCommandMapping.get().keySet()));
+    params.put(DOCUMENTATION_LINK_FIELD, DOCUMENTATION_LINK);
+    params.put(COMMON_COMMANDS, new ArrayList<>(commonPluginCommandMapping.get().keySet()));
+    return params;
+  }
+
+  @Override
+  public CommonPluginCommand getCommonCommand(String commandName) {
+    return commonPluginCommandMapping.get().get(commandName);
+  }
+
+  @Override
+  public PluginCommand getIntegrationCommand(String commandName) {
+    return pluginCommandMapping.get().get(commandName);
+  }
+
+  private Map<String, PluginCommand> getCommands() {
+    HashMap<String, PluginCommand> pluginCommands = new HashMap<>();
+    return pluginCommands;
+  }
+
+  private Map<String, CommonPluginCommand<?>> getCommonCommands() {
+    HashMap<String, CommonPluginCommand<?>> pluginCommands = new HashMap<>();
+    return pluginCommands;
+  }
+
+  protected ObjectMapper configureObjectMapper() {
+    ObjectMapper om = new ObjectMapper();
+    om.setAnnotationIntrospector(new JacksonAnnotationIntrospector());
+    om.configure(MapperFeature.DEFAULT_VIEW_INCLUSION, true);
+    om.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```

Comment:
please add
`om.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);`
in order to serialize Instant type properly

Reply:
Added